rust/nasl-interpreter/src/scheduling/mod.rs

```diff
@@ -284,40 +284,49 @@ where
         {
             let params: Option<Vec<models::Parameter>> =
                 scan.vts.get(i).map(|x| x.parameters.clone());
-            unresolved_dependencies.extend(
+            unknown_dependencies.extend(
                 x.dependencies
                     .iter()
                     .map(|x| storage::Field::NVT(storage::item::NVTField::FileName(x.to_string()))),
             );
             vts.push((x.clone(), params));
         }
 
-        while !unresolved_dependencies.is_empty() {
-            unresolved_dependencies = {
+        while !unknown_dependencies.is_empty() {
+            let new_unresolved_dependencies = {
                 let mut ret = Vec::new();
                 for x in self
-                    .retrieve_by_fields(unresolved_dependencies, storage::Retrieve::NVT(None))?
+                    .retrieve_by_fields(unknown_dependencies, storage::Retrieve::NVT(None))?
                     .filter_map(|(_, f)| match f {
                         storage::Field::NVT(storage::item::NVTField::Nvt(x)) => Some(x),
                         _ => None,
                     })
                 {
                     let stage = Stage::from(&x);
                     tracing::trace!(?stage, oid = x.oid, "adding script_dependency");
-                    ret.extend(x.dependencies.iter().map(|x| {
-                        storage::Field::NVT(storage::item::NVTField::FileName(x.to_string()))
-                    }));
+                    ret.extend(
+                        x.dependencies
+                            .iter()
+                            .filter(|x| !found_dependencies.contains_key(*x))
+                            .map(|x| {
+                                storage::Field::NVT(storage::item::NVTField::FileName(
```

Comment:
Just out of curiosity: Is there a particular reason why you qualify these (`storage::item::NVTField`) instead of importing them via `use` or just a matter of taste?

Reply:
no, I find it sometimes easier to identify something when I am looking for something when I type in the full qualified name

Reply:
I always just let rust-analyzers `replace qualified path with use` code action do its job afterwards